pkg/controllers/user/logging/utils/embedded.go

```diff
@@ -201,6 +207,75 @@ func RemoveEmbeddedTarget(dep rv1beta2.DeploymentInterface, sa rv1.ServiceAccoun
 	return nil
 }
 
+func GetEmbeddedESEndpoint(podLister rv1.PodLister, serviceLister rv1.ServiceLister) (esEndpoint string, err error) {
+	selector := labels.NewSelector()
+	requirement, err := labels.NewRequirement(loggingconfig.LabelK8sApp, selection.Equals, []string{loggingconfig.EmbeddedESName})
+	if err != nil {
+		return "", err
+	}
+
+	espods, err := podLister.List(loggingconfig.LoggingNamespace, selector.Add(*requirement))
+	if err != nil {
+		return "", err
+	}
+	esservice, err := serviceLister.Get(loggingconfig.LoggingNamespace, loggingconfig.EmbeddedESName)
+	if err != nil {
+		return "", err
+	}
+
+	if len(esservice.Spec.Ports) == 0 {
+		return "", fmt.Errorf("get service %s node port failed", loggingconfig.EmbeddedESName)
+	}
+	var esPort int32
+	for _, v := range esservice.Spec.Ports {
+		if v.Name == "http" {
+			esPort = v.NodePort
+			break
+		}
+	}
+
+	if len(espods) == 0 {
+		return "", fmt.Errorf("deploying %s", loggingconfig.EmbeddedESName)
+	}
+	espod := espods[0]
+	if espod.Status.Phase == running {
+		esEndpoint = fmt.Sprintf("http://%s:%v", espod.Status.HostIP, esPort)
+	} else {
+		esEndpoint = fmt.Sprintf("got embedded elasticsearch pod status %s", espod.Status.Phase)
```

Comment:
This will be returned and set as `cl.Spec.EmbeddedConfig.ElasticsearchEndpoint` which makes no sense to me, why not return an error instead.